src/deepsparse/server/config.py

```diff
@@ -12,152 +12,73 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-"""
-Configurations for serving models in the DeepSparse inference server
-"""
 
-import json
-import os
-from functools import lru_cache
-from typing import List
+from typing import List, Optional, Tuple, Union
 
-import yaml
 from pydantic import BaseModel, Field
 
-from deepsparse import PipelineConfig
-from deepsparse.cpu import cpu_architecture
-
 
 __all__ = [
-    "ENV_DEEPSPARSE_SERVER_CONFIG",
-    "ENV_SINGLE_PREFIX",
     "ServerConfig",
+    "EndpointConfig",
+    "SequenceLengthsConfig",
+    "ImageSizesConfig",
 ]
 
 
-ENV_DEEPSPARSE_SERVER_CONFIG = "DEEPSPARSE_SERVER_CONFIG"
-ENV_SINGLE_PREFIX = "DEEPSPARSE_SINGLE_MODEL:"
+class SequenceLengthsConfig(BaseModel):
+    sequence_lengths: List[int] = Field(
+        description="The sequence lengths the model should accept"
+    )
 
 
-class ServerConfig(BaseModel):
-    """
-    A configuration for serving models in the DeepSparse inference server
-    """
-
-    models: List[PipelineConfig] = Field(
-        default=[],
-        description=(
-            "The models to serve in the server defined by PipelineConfig objects"
-        ),
+class ImageSizesConfig(BaseModel):
+    image_sizes: List[Tuple[int, int]] = Field(
+        description="The list of image sizes the model should accept"
     )
-    workers: str = Field(
-        default=max(1, cpu_architecture().num_available_physical_cores // 2),
-        description=(
-            "The number of maximum workers to use for processing pipeline requests. "
-            "Defaults to the number of physical cores on the device."
-        ),
+
+
+class EndpointConfig(BaseModel):
+    name: str = Field(
```

Comment:
would also prefer to make optional - also what's relationship between name and route